### PR TITLE
Computation for perceptual speed Index

### PR DESCRIPTION
**Pull request by kakumar**
Parvez Ahammad (https://github.com/pahammad) and Karan Kumar (https://github.com/kakumar)

### CODE REVIEW

File: visualmetrics.py [hunk 1]
```diff
@@ -36,11 +36,12 @@
 import shutil
 import subprocess
 import tempfile
+from ssim import compute_ssim
```
Comment (pmeenan) [inline visualmetrics.py]: Can we do this at the time of the actual ssim calculation so the dependency becomes a soft dependency and only if ssim-based SI is requested?

Comment (kakumar) [inline visualmetrics.py]: Makes sense. Will do.

File: visualmetrics.py [hunk 2]
```diff
@@ -644,7 +654,7 @@ def calculate_frame_progress(histogram, start, final):
                     target -= this_match
         total += channel_total
         matched += channel_matched
-    progress = (float(matched) / float(total)) if total else 1
+    progress = (float(matched) / float(total))
```
Comment (pmeenan) [inline visualmetrics.py]: Any reason you got rid of the divide-by-zero protection?

Comment (kakumar) [inline visualmetrics.py]: No reason. Will fix.

File: visualmetrics.py [hunk 3]
```diff
@@ -793,16 +841,18 @@ def main():
...
+                if options.orange and \
+                        not os.path.isfile(orange_file):
                     orange_file = os.path.join(temp_dir, 'orange.png')
                     generate_orange_png(orange_file)
                 video_to_frames(options.video, directory, options.force, orange_file, options.viewport, options.full,
                                 options.timeline)
             calculate_histograms(directory, histogram_file, options.force)
-            if options.dir is not None and options.quality is not None:
+            # Perceptual SI computation works on png's only
+            if options.dir is not None and options.quality is not None and options.perceptual is False:
```
Comment (pmeenan) [inline visualmetrics.py]: What about the case where video is provided so png's are created for intermediate processing but jpeg is the end state?  The Perceptual SI should be able to operate on the intermediate png's but still allow final step conversion to jpeg.

Maybe swap the order and calculate the visual metrics before jpeg conversion.

**Comment by pmeenan** [comment]
Just a few minor notes but generally looks good.

There are at least 2 different ways to calculate the perceptual SI that I'd like to experiment with:

1) Calculate ssim versus the end frame (what was implemented here)
2) Calculate SSIM for each frame relative to the next.  Sum up the total of all changes and use the aggregate change to any point as the visual completeness.

#2 has a few interesting benefits:
a) It is constantly increasing
b) The relative SSIM between any 2 frames can be stored and SI can be calculated between any two arbitrary points.

**Pull request merged**